optimum/exporters/neuron/config.py

```diff
@@ -53,3 +56,39 @@ class TextAndVisionNeuronConfig(NeuronConfig):
     """
 
     DUMMY_INPUT_GENERATOR_CLASSES = (DummyTextInputGenerator, DummyVisionInputGenerator, DummyBboxInputGenerator)
+
+
+class DecoderNeuronConfig(ExportConfig):
+    """
+    Handles decoder architectures.
+    """
+
+    NEURONX_ARGS = {"batch_size": 1, "n_positions": 128, "tp_degree": 2, "amp": "f32"}
```

Comment:
Why are those values hardcoded here?

Reply:
These are the default values for these parameters.

Note that for now every single model class in transformers_neuronx has a slightly different set of parameters, but these ones seem to be common. If this is confirmed that these parameters are really common to all models, then I could have declared them directly as named parameters in the _from_transformers method.

Reply:
For me it could change, for instance `tp_degree` will need to be bigger for very big models and so on.

Reply:
If let's say gpt2 and llama not sharing the same fixed non-customizable parameters, may we put it under `GPT2NeuronConfig` directly?

Reply:
Okay, this deserves a better explanation (I was very confused my self when i went through the _transformers-neuronx_ "export").

Firstable, one needs to understand that unlike other models, the export requires a preliminary "modeling/transform" step that happens before the tracing/compilation step.

In the current _transformers_neuronx_ implementation, this step must be performed every time from the source model, even when reloading a model that has already been compiled (this is why I need to copy the "checkpoint" model along the compiled model).

Unfortunately, this step requires the exact same parameters to be passed when the model is first compiled, then reloaded, and believe it or not, the information is not serialized. This is why I need to:
- list the parameters that need to be passed,
- serialize them myself in neuron_config,
- provide my own default values to avoid having one parameter set with a default value by the _transformers_neuronx_ code (note that I actually could have read back the values from the neuronx model `config` member: the one that is not serialized).

Reply:
> If let's say gpt2 and llama not sharing the same fixed non-customizable parameters, may we put it under `GPT2NeuronConfig` directly?

As explained in my previous answer, the parameters CAN be customized, but I need to provide my own default values. But yes, I could have put them under `GPT2NeuronConfig` directly.

Alternatively, as I already mentioned in another answer, I could at least declare batch_size, tp_degree and amp directly in the _from_transformers signature. These are really common to all models.

I thought n_positions was too, but I just double-checked and it is more used as a way to override the original config.n_positions.

Reply:
Done